Filling project summaries

taoTimTim opened the pull request:
<!-- 
Thank you for contributing! Please fill out this template to help us review your PR.
-->

## 📝 Description

This PR completes the project summaries by taking all the data extracted from the various analyses in the system and storing the necessary information in each projects project summary, which is stored in the table `project_summaries`.

Last week there were still certain analyses being done, now that the are all merged the project summaries can be filled, as is done in this PR.

By filling in the project summaries we can properly use the summaries to rank projects, retrieve portfolio information, retrieve resume infomation, and finish the last retrieval requirements in Milestone 1.

New tests have been implemented to test the new functions, and all tests are passing.

**Closes:** #266 

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [ ] ♻️ Refactoring
- [ ] ⚡ Performance improvement

---

## 🧪 Testing

- Ran `pytest`, everything is passing
- Ran the main system and ensured nothing broke and everything wasbeing stored in the database properly:
<img width="1888" height="851" alt="image" src="https://github.com/user-attachments/assets/aac745c2-6906-49a7-861c-dec6ad4ce48b" />


---

## ✓ Checklist

- [x] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [ ] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [x] 🔗 Any dependent changes have been merged and published in downstream modules
- [ ] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots

Screenshot of project summaries after running main system:
<img width="1888" height="851" alt="image" src="https://github.com/user-attachments/assets/5b4c33ff-ddb4-4273-a042-65f226313af2" />

<!-- Add your screenshots here -->

</details>

johaneshp reviewed (approved):
Thanks for working on this timmi! all tests passed. Skills and metrics are loaded to the summary object, and summaries are stored to the database properly. One thing i noticed is that the project summary for text does not contain the activity type, which does not make the result format of project summary uniform between both text and code. I suggest adding the activity_type for text also in the project summary. Other than that, everything works well. Good job timmi!

taoTimTim commented:
@ammaarkhan good catch, just removed the duplicated imports.

@johaneshp Thanks for catching that, I added it into the summary, using a helper function. Pleaase review again.

AdaraPutri reviewed (changes requested):
hey Timmi, thanks for working on this! most metrics and data are stored properly to project_summaries table. one small change I recommend you to make is to replace the current query in `get_project_skills()` from db/skills.py to this:

```sql
SELECT skill_name, level, score, evidence_json
        FROM project_skills
        WHERE user_id = ? AND project_name = ? AND score > 0
        ORDER BY score DESC
```

currently all the skills are added to `summary_json` in `project_summaries`, even those with the score 0. this will cause inaccuracy when we're printing out the list of skills in the resume/ portfolio, as it would list skills that the user did not demonstrate.

taoTimTim commented:
@AdaraPutri Thanks for catching that, Adara! Just fixed it, please review the changes.
